docs/book/src/topics/crd-changes/v1alpha7-to-v1alpha8.md

```diff
@@ -0,0 +1,58 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*
+
+- [v1alpha7 compared to v1alpha8](#v1alpha7-compared-to-v1alpha8)
+  - [Migration](#migration)
+  - [API Changes](#api-changes)
+    - [`OpenStackCluster`](#openstackcluster)
+      - [Change to externalNetworkID](#change-to-externalnetworkid)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# v1alpha7 compared to v1alpha8
+
+> ⚠️ v1alpha8 has not been released yet.
+
+## Migration
+
+All users are encouraged to migrate their usage of the CAPO CRDs from older versions to `v1alpha8`. This includes yaml files and source code. As CAPO implements automatic conversions between the CRD versions, this migration can happen after installing the new CAPO release.
+
+## API Changes
+
+This only documents backwards incompatible changes. Fields that were added to v1alpha8 are not listed here.
+
+### `OpenStackCluster`
+
+#### Change to externalNetworkID
+
+The field `externalNetworkID` has been renamed to `externalNetwork` and isn't a string anymore but a reference to a `NetworkFilter` object.
+
+```yaml
+externalNetworkID: "e60f19e7-cb37-49f9-a2ee-0a1281f6e03e"
+```
+
+becomes
+
+```yaml
+externalNetwork:
+  id: "e60f19e7-cb37-49f9-a2ee-0a1281f6e03e"
+```
+
+It is now possible to specify a `NetworkFilter` object to select the external network to use for the cluster. The `NetworkFilter` object allows to select the network by name, by ID or by tags.
+
+```yaml
+externalNetwork:
+  name: "public"
+```
+
+If a network is provided, it'll be added to `OpenStackCluster.Status.ExternalNetwork`. If the network can't be found, an error will be returned.
+If no network is provided, CAPO will try to find network marked "External" and add it to `OpenStackCluster.Status.ExternalNetwork`. If it can't find a network marked "External",
+`OpenStackCluster.Status.ExternalNetwork` will be set to nil.
```

Comment:
Ah, okay, this is the reason for the new `disableExternalNetwork` knob.

Given we're introducing the ability to filter external networks by name as well as a new API version, could we drop this feature and instead instead users specify an network (or omit it). This warrants a big warning in the migration docs since it'll be a larger (though hopefully easy to spot) change.

Reply:
I'm fine with it. Waiting for @mdbooth's feedback and if he agrees I can propose this change.

Reply:
I don't have any hard data, but I suspect there are a bunch of folks who like the default. Removing it requires the user to be explicit about something that was previously implicit, so we wouldn't be able to convert it.

An alternative might be instead of `disableExternalNetwork` to have `useDefaultExternalNetwork`, i.e. you have to opt-in to the current default.

The disableExternalNetwork knob also allows folks with multiple external networks to not use any of them, which isn't currently possible.

I don't have a strong opinion, tbh. I'm ok with @EmilienM's version.

Soliciting opinions from @lentzi90 @mnaser @mkjpryor.

Reply:
Personally, I quite like the fact that if there is exactly one external network then that gets used without further intervention.

I do love the fact that I will now be able to select my external network by Neutron tag now though, like I can with my cluster network.

Reply:
I agree with @mkjpryor , this is convenient and nice